src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -555,6 +555,9 @@ void QgsRelationReferenceWidget::init()
 
     // Only connect after iterating, to have only one iterator on the referenced table at once
     connect( mComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsRelationReferenceWidget::comboReferenceChanged );
+
+    mReferencedLayer->getFeatures( mComboBox->currentFeatureRequest() ).nextFeature( mFeature );
```

Comment:
Having unconditional I/O in `init()` looks like a potential performance killer, a solution that works with the feedback from the worker thread would be preferrable.

Reply:
I understand this input and I'm looking for a better solution. 
But when we remove this part, shouldn't the following line `updateAttributeEditorFrame( mFeature );` be removed as well? Because the `mFeature` is not valid? Or how was that supposed to work?